pydantic/fields.py

```diff
@@ -261,7 +262,12 @@ def _prep_vals(v_funcs: Iterable[AnyCallable]) -> 'ValidatorsList':
         return [make_generic_validator(f) for f in v_funcs if f]
 
     def validate(
-        self, v: Any, values: Dict[str, Any], *, loc: 'LocType', cls: Optional[Type['BaseModel']] = None
+        self,
+        v: Any,
+        values: Dict[str, Any],
+        *,
+        loc: 'LocType',
+        cls: Optional[Type[Union['BaseModel', 'DataclassType']]] = None,
```

Comment:
perhaps you could use a named type `Optional[Type[Union['BaseModel', 'DataclassType']]` as it's reused in quite a few places.

Reply:
Created a new `ModelType` in `types.py` is that ok?